### PR TITLE
DocLevel Monitor - generate findings when 0 triggers

### PR DESCRIPTION
**Pull request by petardz**
*Issue #, if available:*

*Description of changes:*

DocLevel Monitors generate findings when there's no triggers defined

*CheckList:*
[ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt [hunk 1]
```diff
@@ -210,17 +210,27 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
...
-                docsToQueries,
-                queryToDocIds,
-                dryrun
-            )
+        // If there are no triggers defined, we still want to generate findings
+        if (monitor.triggers.isEmpty()) {
+            if (dryrun == false && monitor.id != Monitor.NO_ID) {
+                docsToQueries.forEach {
+                    val triggeredQueries = it.value.map { queryId -> idQueryMap[queryId]!! }
+                    createFindings(monitor, monitorCtx, triggeredQueries, it.key, true)
```
Comment (eirsep) [inline alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt:218]: why are we even creating findings inside this method?
If we are trying to decouple findings from triggers, why not create findings when they are available?
Do we want to ensure that we don't create findings if triggers fail or something?

**Review by eirsep — CHANGES_REQUESTED** [review]

File: alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt [hunk 2]
```diff
@@ -311,8 +370,7 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
             "test_field.some_other_field" : "us-west-2",
             "f1.type.f2.f3" : "world",
             "f1.type.f4" : "hello",
-            "type" : "some type",
-            "properties": 123
+            "type" : "some type"
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt:373]: why are we changing this?

File: alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt (hunk 1 above)
Comment (lezzago) [reply alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt:218]: If we are not refactoring this now, can you link the github issue to follow up on this in the reply to this comment?

**Comment by petardz** [comment]
Follow-up issue: #865

File: alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt (hunk 1 above)
Comment (petardz) [reply alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt:218]: #865

**Review by lezzago — APPROVED** [review]

**Review by eirsep — APPROVED** [review]

**Pull request merged**